nix/pkgs/purty-pre-commit/default.nix

```diff
@@ -0,0 +1,8 @@
+{ writeScriptBin, purty }:
+
+writeScriptBin "purty" ''
+  #!/usr/bin/env bash
+  for f in "$@"; do
+    ${purty}/bin/purty validate $f
```

Comment:
@gilligan Does this call purty N times, one for each changed file? Isn't it cheaper to call it for the whole project?

Reply:
well it does that because:
- ..this is really what is *supposed* to be happening. The commit hook receives changed files and passes those on to a tool. In this case purty. It should not be checking anything else, definitely not the "whole project" (which it has no concept of anyway though).
- ..purty doesn't support receiving multiple paths on the command line
- ..the overhead of calling purty _[number of files changed]_ times instead of 1 is (fortunately) negligible